tracer/src/Datadog.Trace/Agent/DiscoveryService/DiscoveryService.cs

```diff
@@ -42,10 +44,18 @@ private DiscoveryService(IApiRequestFactory apiRequestFactory)
 
         public string AgentVersion { get; private set; }
 
-        public static DiscoveryService Create(IApiRequestFactory apiRequestFactory)
+        public static DiscoveryService Create(ImmutableExporterSettings exporterSettings)
         {
             lock (GlobalLock)
             {
+                var apiRequestFactory = AgentTransportStrategy.Get(
+                    exporterSettings,
+                    productName: "discovery",
+                    tcpTimeout: null,
+                    AgentHttpHeaderNames.DefaultHeaders,
```

Comment:
You probably don't want/need all these headers. This sends headers like `Datadog-Client-Computed-Top-Level` etc. This request is blocking for all sorts of things so we want to make the requests as lean as possible I think! 🙂

Reply:
Yes, you are right.
As discussed, changed to minimal headers. 360b832aff3dc02a2f871a46cb13bf52d25d75fe